crates/agglayer-aggregator-notifier/src/certifier/mod.rs

```diff
@@ -237,95 +230,45 @@ where
-                        ErrorKind::UnableToExecuteProver => {
-                            CertificationError::InternalError("Unable to execute prover".into())
-                        }
-                        ErrorKind::ProverFailed => {
-                            CertificationError::ProverFailed(source_error.message().to_string())
-                        }
-                        ErrorKind::ProofVerificationFailed => {
-                            let proof_error: Result<
-                                pessimistic_proof::error::ProofVerificationError,
-                                _,
-                            > = bincode::default().deserialize(&error.error);
-
-                            match proof_error {
-                                Ok(error) => {
-                                    CertificationError::ProofVerificationFailed { source: error }
-                                }
-                                Err(_source) => {
-                                    warn!(
-                                        "Failed to deserialize the error details coming from the \
-                                         prover: {source_error:?}"
-                                    );
-
-                                    CertificationError::InternalError(
-                                        source_error.message().to_string(),
-                                    )
-                                }
-                            }
-                        }
-
-                        ErrorKind::ExecutorFailed => {
-                            let proof_error: Result<pessimistic_proof::ProofError, _> =
-                                bincode::default().deserialize(&error.error);
-
-                            match proof_error {
-                                Ok(error) => {
-                                    CertificationError::ProverExecutionFailed { source: error }
-                                }
-                                Err(_source) => {
-                                    warn!(
-                                        "Failed to deserialize the error details coming from the \
-                                         prover: {source_error:?}"
-                                    );
-
-                                    CertificationError::InternalError(
-                                        source_error.message().to_string(),
-                                    )
-                                }
-                            }
-                        }
-                        ErrorKind::Unspecified => {
-                            CertificationError::InternalError(source_error.message().to_string())
-                        }
-                    }
+                if let Some(error) = source_error.downcast_ref::<prover_executor::Error>() {
+                    error!("Failed to generate proof: {}", error);
+
+                    CertificationError::ProverFailed(error.to_string())
```

Comment:
Should we keep the source `prover_executor` error here as part of `ProverFailed`, now that we have the type and details?